Library/Homebrew/test/support/helper/env.rb

```diff
@@ -1,13 +1,31 @@
 module Test
   module Helper
     module Env
+      def copy_env
+        ENV.to_hash
+      end
+
+      def restore_env(env)
+        ENV.replace(env)
+      end
+
       def with_environment(partial_env)
-        old = ENV.to_hash
+        old = copy_env
         ENV.update partial_env
-        begin
+        yield
+      ensure
+        restore_env old
+      end
+
+      def using_git_env
```

Comment:
All green apart from the `brew bundle` integration test, because `brew bundle` [requires](https://github.com/Homebrew/homebrew-bundle/blob/575804057c7195f647d747683cc4c1592f28417a/cmd/brew-bundle.rb#L36) a commit on or after `Wed Aug 17 11:07:17 2016 +0100`.

So, three options here:

1. Give up on removing this method
2. Delete the `GIT_COMMITTER_DATE` environment variable before this test.
3. Update the date used so it's more recent, and also update all of the expected commit hashes that rely on the old test.

I think I'd probably prefer 3. What do you think?